### PR TITLE
[NCL-7230] Implement Product Detail page

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/ProductDetailPage/ProductDetailPage.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  useTitle(generatePageTitle({ serviceContainer: serviceContainerProduct, firstLevelEntity: 'Product' }));
+
+  const attributes = [
+    { name: 'Abbreviation', value: serviceContainerProduct.data?.abbreviation },
+    { name: 'Product Managers', value: serviceContainerProduct.data?.productManagers },
+    { name: 'Product Pages Code', value: serviceContainerProduct.data?.productPagesCode },
+  ];
+
+  return (
+    <ServiceContainerLoading {...serviceContainerProduct} title="Project details">
```
Comment (matedo1) [inline src/components/ProductDetailPage/ProductDetailPage.tsx:55]: `title="Product details"`

**Review by matedo1 — CHANGES_REQUESTED** [review]

**Comment by matedo1** [comment]
Hi guys @patrikk0123 and @DnsZhou , 

`components/ProductDetailPage/ProductDetailPage.tsx` contains bug, would be able to found it? :)

File: src/services/productsApi.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { AxiosRequestConfig } from 'axios';
+
+import { Product, ProductVersionPage } from 'pnc-api-types-ts';
+
+import { pncClient } from './pncClient';
+
+interface IProductApiData {
+  id: string;
+}
+
```
Comment (matedo1) [inline src/services/productsApi.tsx:10]: wrong file name:
1) use singular
2) use `ts` 

so `productApi.ts`

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/components/ProductDetailPage/ProductDetailPage.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  const serviceContainerProductVersions = useServiceContainer(productsApi.getProductVersions);
+  const serviceContainerProductVersionsRunner = serviceContainerProductVersions.run;
+
+  useEffect(() => {
+    serviceContainerProductRunner({ serviceData: { id: productId } });
+  }, [serviceContainerProductRunner, productId]);
+
+  useQueryParamsEffect(
+    ({ requestConfig } = {}) => serviceContainerProductVersionsRunner({ serviceData: { id: productId }, requestConfig }),
```
[the remark this answers is not in the document]
Comment (patrikk0123) [reply src/components/ProductDetailPage/ProductDetailPage.tsx:41]: No, it is not, because of typescript error.

**Review by matedo1 — APPROVED** [review]

**Pull request merged**